### PR TITLE
Fixed install/remove of influxdb on non-systemd Debian/Ubuntu systems

### PR DESCRIPTION
**Pull request by martinseener**
See PR on telegrad for rationale: https://github.com/influxdata/telegraf/pull/2360

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

**Comment by martinseener** [comment]
Also updated package.sh Go version to 1.7.5 which is an addition to #7926 and also does the same as #7947 (so this can be closed and the contributor doesn't need to sign the CLA)

**Comment by martinseener** [comment]
@jwilder removed package.sh with latest rebase

**Comment by martinseener** [comment]
@jwilder i've moved that one to 1.3.0 as well. I really hope that you or some other maintainer is merging this soon as it is a bug and peeving not just us here at work 🤕

File: scripts/post-install.sh [hunk 1]
```diff
@@ -45,29 +44,42 @@ fi
...
+    if [[ "$(readlink /proc/1/exe)" == */systemd ]]; then
+        install_systemd
     else
-	# Assuming sysv
-	install_init
-	install_update_rcd
+        # Assuming SysVinit
+        install_init
+        # Run update-rc.d or fallback to chkconfig if not available
+        if which update-rc.d >/dev/null; then
```
Comment (rossmcdonald) [inline scripts/post-install.sh]: All calls to `which` should be updated with `&>/dev/null`, not `>/dev/null`. On CentOS, I get this on a vanilla installation:

```
$ sudo rpm -i influxdb-1.2.1~rc3~cab47cd.x86_64.rpm
which: no update-rc.d in (/sbin:/bin:/usr/sbin:/usr/bin:/usr/X11R6/bin)
which: no update-rc.d in (/sbin:/bin:/usr/sbin:/usr/bin:/usr/X11R6/bin)
```

File: scripts/post-install.sh [hunk 2]
```diff
@@ -45,29 +44,42 @@ fi
...
 elif [[ -f /etc/os-release ]]; then
     source /etc/os-release
     if [[ $ID = "amzn" ]]; then
-	# Amazon Linux logic
-	install_init
-	install_chkconfig
+        # Amazon Linux logic
+        install_init
+        # Run update-rc.d or fallback to chkconfig if not available
+        if which update-rc.d >/dev/null; then
```
Comment (rossmcdonald) [inline scripts/post-install.sh]: Should be updated to `&>/dev/null`

File: scripts/post-install.sh (hunk 1 above)
Comment (martinseener) [inline scripts/post-install.sh]: fixed

File: scripts/post-install.sh (hunk 2 above)
Comment (martinseener) [inline scripts/post-install.sh]: fixed. also there was a third place in this file which i also fixed ;)

**Comment by martinseener** [comment]
I don't want to be crude but can anybody tell me please, why this hasn't been merged into 1.3.0? Is something missing or not working (i've tested this one extensively on several distros with either SysVinit and Systemd)
In Telegraf, this has already been merged and released in 1.3.0 (https://github.com/influxdata/telegraf/pull/2360).

**Comment by martinseener** [comment]
Why won't fix? If i repair the branch again, will it then be merged after more than 2 years?

**Comment by martinseener** [comment]
Any updates on this? Can anyone of the reviewers @rossmcdonald @gunnaraasen say something about this? It's still an issue for us and not merged after 2 years. It's a shame. Please give some information wether it's planned to be merged or not. But please not just ignore it for a long time and let a bot close it. Thanks alot!

**Pull request closed without merging**